aws/resource_aws_network_interface.go

```diff
@@ -165,25 +165,36 @@ func resourceAwsNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) e
 	}
 
 	eni := describeResp.NetworkInterfaces[0]
-	d.Set("subnet_id", eni.SubnetId)
-	d.Set("private_ip", eni.PrivateIpAddress)
+
+	attachment := []map[string]interface{}{}
+
+	if eni.Attachment != nil {
+		attachment = []map[string]interface{}{flattenAttachment(eni.Attachment)}
+	}
+
+	if err := d.Set("attachment", attachment); err != nil {
+		return fmt.Errorf("error setting attachment: %s", err)
+	}
+
+	d.Set("description", eni.Description)
 	d.Set("private_dns_name", eni.PrivateDnsName)
-	d.Set("private_ips", flattenNetworkInterfacesPrivateIPAddresses(eni.PrivateIpAddresses))
-	d.Set("security_groups", flattenGroupIdentifiers(eni.Groups))
-	d.Set("source_dest_check", eni.SourceDestCheck)
+	d.Set("private_ip", eni.PrivateIpAddress)
 
-	if eni.Description != nil {
-		d.Set("description", eni.Description)
+	if err := d.Set("private_ips", flattenNetworkInterfacesPrivateIPAddresses(eni.PrivateIpAddresses)); err != nil {
+		return fmt.Errorf("error setting private_ips: %s", err)
 	}
 
-	// Tags
-	d.Set("tags", tagsToMap(eni.TagSet))
+	d.Set("private_ips_count", len(eni.PrivateIpAddresses)-1)
```

Comment:
Why the need to subtract 1 from the total count?

Reply:
Information documented on https://www.terraform.io/docs/providers/aws/r/network_interface.html#private_ips_count :+1: